sympy/calculus/util.py

```diff
@@ -112,3 +119,672 @@ def elm_domain(expr, intrvl):
+                return True
+            if self.min > other:
+                return False
+        return None
+
+    def __gt__(self, other):
+        """
+        Returns True if range of values attained by `self` Limits object
+        is greater than the range of values attained by `other`, where other
+        may be any value of type Limits object or extended real number value,
+        , False is returned if `other` satisfies the same property, None if
+        the values attained by Limits objects intersect.
+
+        Examples
+        ========
+
+        >>> from sympy import Limits, oo
+        >>> Limits(1, 3) > Limits(4, oo)
+        False
+        >>> Limits(1, 4) > Limits(3, 4)
+
+        >>> Limits(1, oo) > -1
+        True
+
+        """
+        other = _sympify(other)
+        if isinstance(other, Limits):
+            if self.min > other.max:
+                return True
+            if self.max <= other.min:
+                return False
+            return
+
+        if not (other.is_real or other is S.Infinity or other is S.NegativeInfinity):
+            raise TypeError("Invalid comparison of %s %s" % (type(other), other))
+
+        if other.is_comparable:
+            if self.min > other:
+                return True
+            if self.max <= other:
+                return False
+        return None
+
+    def __ge__(self, other):
+        """
+        Returns True if range of values attained by `self` Limits object
+        is less that the range of values attained by `other`, where other
+        may be any value of type Limits object or extended real number value,
+        False is returned if `other` satisfies the same property, None if
+        the values attained by Limits objects intersect.
+
+        Examples
+        ========
+
+        >>> from sympy import Limits, oo
+        >>> Limits(1, 3) >= Limits(4, oo)
+        False
+        >>> Limits(1, 4) >= Limits(3, 4)
+
+        >>> Limits(1, oo) >= 1
+        True
+
+        """
+        other = _sympify(other)
+        if isinstance(other, Limits):
+            if self.min >= other.max:
+                return True
+            if self.max < other.min:
+                return False
+            return None
+
+        if not (other.is_real or other is S.Infinity or other is S.NegativeInfinity):
+            raise TypeError("Invalid comparison of %s %s" % (type(other), other))
+
+        if other.is_comparable:
+            if self.min >= other:
+                return True
+            if self.max < other:
+                return False
+        return None
+
+    def __contains__(self, other):
+        """
+        Returns True if other is contained in self, where other
+        belongs to extended real numbers, False if not contained,
+        otherwise TypeError is raised.
+
+        Examples
+        ========
+
+        >>> from sympy import Limits, oo
+        >>> 1 in Limits(-1, 3)
+        True
+        >>> -oo in Limits(1, oo)
```

Comment:
This should be changed. One should always think that ``-oo`` and ``oo`` go together as limits.